src/test/common/test_crc32c.cc

```diff
@@ -265,3 +265,44 @@ TEST(Crc32c, RangeNull) {
     ASSERT_EQ(crc, *check);
   }
 }
+
+TEST(Crc32c, zeros_performance_compare) {
+  utime_t start;
+  utime_t end;
+
+  for (size_t scale=1; scale < 31; scale++)
+  {
+    size_t size = (1<<scale) + rand()%(1<<scale);
+    start = ceph_clock_now();
+    uint32_t crc_a = ceph_crc32c(111, nullptr, size);
+    end = ceph_clock_now();
+    std::cout << "regular  method. size=" << size << " time= " << (double)(end-start)
+        << " at " << (double)size/(1024*1024)/(double)(end-start) << " MB/sec" << std::endl;
+
+    start = ceph_clock_now();
+    uint32_t crc_b = ceph_crc32c_sctp(111, nullptr, size);
+    end = ceph_clock_now();
+    std::cout << "fallback method. size=" << size << " time=" << (double)(end-start)
+        << " at " << (double)size/(1024*1024)/(double)(end-start) << " MB/sec" << std::endl;
+
+    EXPECT_EQ(crc_a, crc_b);
+  }
+}
+
+TEST(Crc32c, zeros_performance) {
+  constexpr size_t ITER=100000;
+  utime_t start;
+  utime_t end;
+
+  start = ceph_clock_now();
+  for (size_t i=0; i<ITER; i++)
+  for (size_t scale=1; scale < 31; scale++)
+  {
+    size_t size = (1<<scale) + rand() % (1<<scale);
+    ceph_crc32c(rand(), nullptr, size);
```

Comment:
Here you are using rand twice: once to excercise the case where size is not on a power of 2 boundary, then a second time to randomize the input crc value.  I'm fine with this because over the course of 100,000 iterations the input crc values should not significantly impact the time it takes.  

The only reason I mention it is because @branch-predictor was talking about unpredictable results.  I wasn't sure if he was objecting to the use of rand (and not just the use of low resolution timers).  

Note that if you try to call 1,000 iterations (let alone 100,000 iterations) of ceph_crc32c_func instead of ceph_crc32c_zeros, with a constant buffer size of 1GB it takes several minutes to complete but with your algorithm it takes seconds.  More evidence of how much faster your algorithm is than the previously existing methods.